components/raftstore/src/store/fsm/store.rs

```diff
@@ -2213,86 +2222,24 @@ impl<'a, EK: KvEngine, ER: RaftEngine, T: Transport> StoreFsmDelegate<'a, EK, ER
-        let schedule_gc_snap = |region_id: u64, snaps| -> Result<()> {
-            debug!(
-                "schedule snap gc";
-                "store_id" => self.fsm.store.id,
-                "region_id" => region_id,
-            );
-
-            let gc_snap = PeerMsg::CasualMessage(CasualMessage::GcSnap { snaps });
-            match self.ctx.router.send(region_id, gc_snap) {
-                Ok(()) => Ok(()),
-                Err(TrySendError::Disconnected(_)) if self.ctx.router.is_shutdown() => Ok(()),
-                Err(TrySendError::Disconnected(PeerMsg::CasualMessage(
-                    CasualMessage::GcSnap { snaps },
-                ))) => {
-                    // The snapshot exists because MsgAppend has been rejected. So the
-                    // peer must have been exist. But now it's disconnected, so the peer
-                    // has to be destroyed instead of being created.
-                    info!(
-                        "region is disconnected, remove snaps";
-                        "region_id" => region_id,
-                        "snaps" => ?snaps,
-                    );
-                    for (key, is_sending) in snaps {
-                        let snap = match self.ctx.snap_mgr.get_snapshot_for_gc(&key, is_sending) {
-                            Ok(snap) => snap,
-                            Err(e) => {
-                                error!(%e;
-                                    "failed to load snapshot";
-                                    "snapshot" => ?key,
-                                );
-                                continue;
-                            }
-                        };
-                        self.ctx
-                            .snap_mgr
-                            .delete_snapshot(&key, snap.as_ref(), false);
-                    }
-                    Ok(())
-                }
-                Err(TrySendError::Full(_)) => Ok(()),
-                Err(TrySendError::Disconnected(_)) => unreachable!(),
-            }
-        };
-        for (key, is_sending) in snap_keys {
-            if last_region_id == key.region_id {
-                keys.push((key, is_sending));
-                continue;
-            }
-
-            if !keys.is_empty() {
-                schedule_gc_snap(last_region_id, keys)?;
-                keys = vec![];
-            }
-
-            last_region_id = key.region_id;
-            keys.push((key, is_sending));
-        }
-        if !keys.is_empty() {
-            schedule_gc_snap(last_region_id, keys)?;
-        }
-        Ok(())
-    }
-
     fn on_snap_mgr_gc(&mut self) {
-        if let Err(e) = self.handle_snap_mgr_gc() {
-            error!(?e;
-                "handle gc snap failed";
+        // refresh multi_snapshot_files enable flag
+        self.ctx.snap_mgr.set_enable_multi_snapshot_files(
+            self.ctx
+                .feature_gate
+                .can_enable(MULTI_FILES_SNAPSHOT_FEATURE),
```

Comment:
Maybe we need a test for this feature gate.

Reply:
> Maybe we need a test for this feature gate.

Good point. Added an integration test.